src/finufft.cpp

```diff
@@ -1112,26 +1132,102 @@ int FINUFFT_EXECUTE(FINUFFT_PLAN p, CPX *cj, CPX *fk) {
       // STEP 1: (varies by type)
       timer.restart();
       if (p->type == 1) { // type 1: spread NU pts p->X, weights cj, to fw grid
-        spreadinterpSortedBatch(thisBatchSize, p, cjb);
+        spreadinterpSortedBatch(thisBatchSize, p, fwBatch, cjb);
         t_sprint += timer.elapsedsec();
       } else { //  type 2: amplify Fourier coeffs fk into 0-padded fw
-        deconvolveBatch(thisBatchSize, p, fkb);
+        deconvolveBatch(thisBatchSize, p, fwBatch, fkb);
         t_deconv += timer.elapsedsec();
       }
 
-      // STEP 2: call the pre-planned FFT on this batch
+      // STEP 2: call the FFT on this batch
       timer.restart();
+#ifdef FINUFFT_USE_DUCC0
```

Comment:
Question: I'm using opts.nthreads in the following, and it if it is >1,but OpenMP is switched off,the code will *still* run in parallel. Do I need to safeguard against this case?

Reply:
Yes, If omp is switched off no parallelization should happen. It might conflict with users that parallelize by hand

Reply:
If I build without openMP I get:

```
libfinufft.so: undefined reference to `pthread_create'
libfinufft.so: undefined reference to `pthread_getaffinity_np'
libfinufft.so: undefined reference to `pthread_setaffinity_np'
libfinufft.so: undefined reference to `pthread_join'
```

I can have CMAKE linking against pthread. I have a question, are you using ptread directly in ducc0? Shall we have this wrapped in ifdefs? if pthread is not linked -> no parallelism? ( we can mode this discussion in ducc0 )

Reply:
OK, I'll take extra care to switch off multithreaing if OpenMP is not active!

I am using C++'s `std::thread` class for multithreading, which should in principle work out of the box. On Linux and MacOS this is built on pthreads.
The best way to get multithreading for ducc compiled in is to use `-pthread` for both compilation and linking; that should take care of everything. Not sure how it works on Windows, but it's possible that no flags are needed at all.

Reply:
OK, Im now using `min<size_t>(MY_OMP_GET_MAX_THREADS(), p->opts.nthreads)` as the number of threads; that should work as expected in all circumtances, as fas as I can see.